### ISSUE TITLE
Include transaction ID in logged messages

### ISSUE THREAD

**Issue by bjustin-ibm** [openwhisk/openwhisk-package-kafka#106]
Right now, logged messages are completely interleaved and it is hard to follow individual events through the system (creating a trigger, responding to a message, etc.). 

If the logged messages used a transaction ID the way the OpenWhisk core system does, it would be a lot easier to follow events through the log.

**Comment by houshengbo** [comment]
For Python, we can use filter to log the transaction information.
One question is that transaction ID is managed as an object, which also other implemented methods besides the id itself, in openwhisk core. Do we need the same approach or just an ID?
When and how do we generate the ID, if it is not something we can directly get?

**Comment by bjustin-ibm** [comment]
This transaction ID will be specific to the feed provider. That is, it will only be used by the feed provider and will be different from any TID created by the OpenWhisk system. Because this feed provider operates independently of the OW core, it cannot possibly obtain a TID generated by the core.

It is at our discretion about when to create new TIDs, but it should be easy to identify these situations when we consider that the purpose of the TID is to de-interleave log messages. So the first rule of TIDs is: You only need a TID when there is a potential to interleave log messages. For a counter example, I don't think that any messages logged by the "doctor" thread need a TID, because the doctor is a single thread, and can't possibly result in interleaved log messages, so long as they are properly identified as being from the doctor - which they already are 😉 

Similarly, the polling loop of each Consumer is already grouped by logging the ID of the trigger. And since each Consumer runs in its own unique thread, it is not possible for messages from the same Consumer to be interleaved with each other. As such, as long as the log message includes the trigger ID, then there is no need to generate a TID for (as an example) each Consumer, or each `poll()` loop of each Consumer.

However, there is a need for a TID when a request to create or delete a trigger comes in. These are clearly separate "transactions" and can be interleaved. All log messages pertaining to the fulfillment of these requests should have a unique, but shared TID until each request is completed.
